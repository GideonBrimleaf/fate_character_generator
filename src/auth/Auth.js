import React, { useEffect, useState } from 'react';
import { auth } from '../lib/fire';

export const AuthContext = React.createContext();

export const AuthProvider = ({ children }) => {
  const [currentUser, setCurrentUser] = useState(null);
  const [pending, setPending] = useState(true);

  useEffect(() => {
    // TODO - this observer is firing and triggering state
    // changes in test renders which can't be handled with "act",
    // this is creating console warnings in test runs
    auth.onAuthStateChanged((user) => {
      setCurrentUser(user)
      setPending(false)
    });
  }, []);

  if (pending) {
    return <>Loading...</>
  }

  return (
    <AuthContext.Provider
      value={{
        currentUser,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
};
