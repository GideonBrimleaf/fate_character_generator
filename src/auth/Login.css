.login-input:-webkit-autofill,
.login-input:-webkit-autofill:hover, 
.login-input:-webkit-autofill:focus,
textarea:-webkit-autofill,
textarea:-webkit-autofill:hover,
textarea:-webkit-autofill:focus,
select:-webkit-autofill,
select:-webkit-autofill:hover,
select:-webkit-autofill:focus {
  -webkit-text-fill-color: white;
  -webkit-box-shadow: 0 0 0px 1000px #282c34 inset;
  font-size: x-large;
}

.login-input::placeholder {
  text-align: left;
  font-size: x-large;
  color: white;
}

.login-input {
  min-height: 2.5vh;
  background-color: #282c34;
  border-top-style: none;
  border-right-style: none;
  border-left-style: none;
  border-bottom-color: white;
  caret-color: white;
  color: white;
  font-size: x-large;
}

.login-input[type="email"]:focus {
  outline: none;
}

.login-input[type="email"]:focus:hover {
  outline: none;
}

.login-input[type="password"]:focus {
  outline: none;
}

.login-input[type="password"]:focus:hover {
  outline: none;
}