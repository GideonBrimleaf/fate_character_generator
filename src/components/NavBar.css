.nav-bar {
  list-style: none;
  display: flex;
  justify-content: space-around;
}

.nav-item {
  margin-right: 1em;
}

.nav-link {
  color: white;
  text-decoration: none;
}