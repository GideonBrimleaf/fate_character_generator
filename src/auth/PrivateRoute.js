import React, { useContext } from "react";
import { Route, Redirect } from "react-router-dom";
import { AuthContext } from "./Auth";

const PrivateRoute = ({ component: RouteComponent, data, ...rest }) => {
  const {currentUser} = useContext(AuthContext);
  return (
    <Route
      {...rest}
      render={routeProps =>
        !!currentUser ? (
          <RouteComponent {...routeProps} {...data} />
        ) : (
          <Redirect to={"/login"} />
        )
      }
    />
  );
};


export default PrivateRoute