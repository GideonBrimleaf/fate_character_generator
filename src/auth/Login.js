import React, { useCallback, useContext } from "react";
import { withRouter, Redirect } from "react-router";
import { auth } from "../fire";
import { AuthContext } from "./Auth.js";
import './Login.css'

const Login = ({ history }) => {
  const handleLogin = useCallback(
    async event => {
      event.preventDefault();
      const { email, password } = event.target.elements;
      try {
        await auth.signInWithEmailAndPassword(email.value, password.value);
        history.push("/");
      } catch (error) {
        alert(error);
      }
    },
    [history]
  );

  const { currentUser } = useContext(AuthContext);

  if (currentUser) {
    return <Redirect to="/" />;
  }

  return (
    <div>
      <h1>Log In</h1>
      <form onSubmit={handleLogin}>
        <input className="login-input" name="email" type="email" placeholder="Email:" />
        <input className="login-input" name="password" type="password" placeholder="Password:" />
        <button className="primary-button" type="submit">Log in</button>
      </form>
    </div>
  );
};

export default withRouter(Login);