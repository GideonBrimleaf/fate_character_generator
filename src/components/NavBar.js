import React from 'react'
import { Link } from 'react-router-dom'
import './NavBar.css'

const NavBar = () => {
  return (
    <ul className="nav-bar">
      <li className="nav-item">
        <Link className="nav-link" to='/'>Home</Link>
      </li>
      <li>
        <Link className="nav-link" to='/New'>New Character</Link>
      </li>
    </ul>
  )
}

export default NavBar
