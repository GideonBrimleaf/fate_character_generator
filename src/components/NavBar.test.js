import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";

import { AuthContext } from '../auth/Auth';
import NavBar from './NavBar';

it('renders correctly when user is not logged in', () => {
  // Arrange
  const props = {}

  // Act
  const result = render(
    <AuthContext.Provider value={{ currentUser: null }}>
      <NavBar {...props} />
    </AuthContext.Provider>
  );

  // Assert
  expect(result.asFragment()).toMatchSnapshot();
});

it('renders correctly when user is logged in', () => {
  // Arrange
  const props = {}

  // Act
  const result = render(
    <AuthContext.Provider value={{ currentUser: true }}>
      <MemoryRouter>
        <NavBar {...props} />
      </MemoryRouter>
    </AuthContext.Provider>
  );

  // Assert
  expect(result.asFragment()).toMatchSnapshot();
});
