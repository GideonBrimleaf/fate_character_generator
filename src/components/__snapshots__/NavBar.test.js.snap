// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when user is logged in 1`] = `
<DocumentFragment>
  <ul
    class="nav-bar"
  >
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        href="/"
      >
        Sign Out
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`renders correctly when user is not logged in 1`] = `<DocumentFragment />`;
